### ISSUE TITLE
[BLOCKER] Implement Automated Hyperparameter Tuning for Optimal Performance

### ISSUE THREAD

**Issue by frankbria** [frankbria/narrative-modeling-app#77]
## 🎯 Overview
Implement automated hyperparameter optimization using grid search, random search, and Bayesian optimization to find optimal model parameters.

## 📊 Current State
- Models trained with default parameters only
- No optimization of hyperparameters
- Suboptimal model performance

## 🎯 Expected Outcome
- Automatic hyperparameter tuning for selected algorithms
- Smart parameter space definition based on data characteristics
- Multiple tuning strategies (grid, random, Bayesian)
- Best parameters automatically applied
- Tuning results visualization

## ✅ Acceptance Criteria
- [ ] Automated hyperparameter tuning for each algorithm
- [ ] Three tuning strategies:
  - Grid search (exhaustive but slow)
  - Random search (faster, good coverage)
  - Bayesian optimization (intelligent, efficient)
- [ ] Smart parameter space definition:
  - Algorithm-specific parameters
  - Data-driven parameter ranges
  - Expert defaults as baseline
- [ ] Tuning configuration:
  - Time budget
  - Number of trials
  - Optimization metric
  - Cross-validation folds
- [ ] Results visualization:
  - Parameter importance
  - Optimization history
  - Best parameters highlight
  - Performance improvement vs. default

### Backend (Python/FastAPI)
```python
class HyperparameterTuner:
    async def tune(
        self,
        algorithm: str,
        X: pd.DataFrame,
        y: pd.Series,
        strategy: str,
        config: TuningConfig
    ) -> TuningResult:
        \"\"\"Run hyperparameter tuning\"\"\"
        pass
    
    async def define_search_space(
        self,
        algorithm: str,
        data_profile: DataProfile
    ) -> Dict[str, Any]:
        \"\"\"Define parameter search space\"\"\"
        pass

class TuningConfig(BaseModel):
    strategy: str  # 'grid' | 'random' | 'bayesian'
    time_budget: int = 600
    n_trials: int = 50
    cv_folds: int = 5
    scoring: str = 'accuracy'

class TuningResult(BaseModel):
    best_params: Dict[str, Any]
    best_score: float
    all_results: List[Dict]
    parameter_importance: Dict[str, float]
    improvement_over_default: float
```

- Use optuna for Bayesian optimization
- Use scikit-learn for grid/random search
- Parallel trial execution
- Early stopping for poor trials

## 📋 Subtasks
- [ ] Implement grid search tuning
- [ ] Implement random search tuning
- [ ] Integrate Bayesian optimization (optuna)
- [ ] Define parameter spaces for each algorithm
- [ ] Create adaptive parameter range selection
- [ ] Build tuning results visualization
- [ ] Add parallel trial execution
- [ ] Implement early stopping
- [ ] Create unit tests for tuning
- [ ] Add integration tests
- [ ] Update documentation

## 🔗 Dependencies
- #75 (Core AutoML engine)

## 🏷️ Labels
\`blocker\`, \`backend\`, \`ml-algorithms\`, \`stage-5-model-training\`

## ⏱️ Estimated Effort
3-4 weeks
